third_party/hdf5.BUILD

```diff
@@ -0,0 +1,363 @@
+'''
+hdf5@1.12.2
+'''
+
+licenses(["notice"])  # BSD-like license
+
+exports_files(["COPYING"])
+
+cc_library(
+    name = "hdf5",
+    srcs = glob(
+        [
+            "src/**/*.c",
+            "hl/src/*.c",
+            "c++/src/*.cpp",
+        ],
+        exclude = [
+            "src/H5make_libsettings.c",
+            "src/H5detect.c",
+        ],
+    ) + [
+        "config/H5Tinit.c",
+        "config/H5lib_settings.c",
+    ],
+    copts = select({
+        "@bazel_tools//src/conditions:windows": [
+            "/Zc:preprocessor-",
+        ],
+        "//conditions:default": [
+            "-Wno-error=implicit-function-declaration",
+        ],
+    }),
+    includes = [
+        "c++/src",
+        "config",
+        "hl/src",
+        "src",
+        "src/H5FDsubfiling",
+    ],
+    linkopts = [],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":header",
+    ],
+)
+
+cc_library(
+    name = "header",
+    srcs = [],
+    hdrs = glob([
+        "src/**/*.h",
+        "hl/src/*.h",
+        "c++/src/*.h",
+    ]) + [
+        "config/H5pubconf.h",
+    ],
+    copts = select({
+        "@bazel_tools//src/conditions:windows": [
+            "/Zc:preprocessor-",
+        ],
+        "//conditions:default": [],
+    }),
+    includes = [
+        "c++/src",
+        "config",
+        "hl/src",
+        "src",
+        "src/H5FDsubfiling",
+    ],
+    linkopts = [],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@zlib",
+    ],
+)
+
+genrule(
+    name = "H5pubconf_h",
+    outs = ["config/H5pubconf.h"],
+    cmd = "\n".join([
+        "cat <<'EOF' >$@",
```

Comment:
These settings all look pretty reasonable.

I think it'll probably be sufficient for all the platforms we're targeting for this library (x64,aarch64 variants of macos and linux).